cli/src/semgrep/meta.py

```diff
@@ -99,7 +99,24 @@ def repo_name(self) -> str:
 
     @property
     def repo_url(self) -> Optional[str]:
-        return get_url_from_sstp_url(os.getenv("SEMGREP_REPO_URL"))
+        env = get_state().env
+        repo_url = os.getenv("SEMGREP_REPO_URL")
+        if not repo_url:
+            # if the repo URL was not explicitly provided, try getting it from git
+            # nosem: use-git-check-output-helper
+            rev_parse = subprocess.run(
+                ["git", "config", "--get", "remote.origin.url"],
+                capture_output=True,
+                encoding="utf-8",
+                timeout=env.git_command_timeout,
+            )
+            if rev_parse.returncode != 0:
+                raise Exception(
+                    "Unable to infer repo_url. Set SEMGREP_REPO_URL environment variable or run in a valid git project"
+                )
```

Comment:
doesn't this just crash if the URL couldn't be extracted?

also, if that's intended, I think there's some specific class we should be using, maybe SemgrepError?

Reply:
good points -- I'm not sure why we're using Exception, but it's consistent across the entire file. I do think that it would be pretty weird to run a CI scan on a file without a remote, so I'm ok with it crashing.

Reply:
@ievans the issue is you're asserting not just that there is a remote, but also that it's named `origin`. I'm not sure how safe that second assumption is.

Reply:
good point -- I'll make it a warning